scripts/extract_pdf_abstract.py

```diff
@@ -107,28 +109,58 @@ def extract(path_pdf):
     # clean up temp file
     os.remove(path_tmp_pdf)
 
-    # TODO: Write to JSON
-    # print(repr(abstract))
+    out = {'@key': key, 'abstract': abstract}
 
+    return out
 
-def main(path):
+
+def main(records, pdf_dir, output_dir, num_cpus=-1, verbose=0):
     """Main function."""
 
-    path_pdfs = glob.glob(os.path.join(path, '*.pdf'))
+    path_pdfs = []
+
+    for cur_key in records.keys():
+        cur_fn = cur_key.split('/')[-1]
+        cur_path = os.path.join(pdf_dir, '{}.pdf'.format(cur_fn))
+        path_pdfs.append((cur_key, cur_path))
 
     dfx = delayed(extract)
-    pool = Parallel(n_jobs=-1, verbose=1)
+    pool = Parallel(n_jobs=num_cpus, verbose=verbose)
+    abstracts = pool(dfx(cur_key, cur_path) for cur_key, cur_path in tqdm.tqdm(path_pdfs))
```

Comment:
does tqdm play nicely with parallel's verbose? i'd expect this to produce pretty interesting logs to stdout

Reply:
Plays nicely together. Of course, it increases #NoOfProcesses per iteration but for long queues this makes sense.

Reply:
![screen shot 2018-11-09 at 19 12 50](https://user-images.githubusercontent.com/7326732/48280290-7c157680-e453-11e8-90fa-546a0d00fa45.png)

fyi